service/matching/physical_task_queue_manager.go

```diff
@@ -126,12 +132,109 @@ type (
 		clusterMeta          cluster.Metadata
 		taggedMetricsHandler metrics.Handler // namespace/taskqueue tagged metric scope
 		// pollerHistory stores poller which poll from this taskqueue in last few minutes
-		pollerHistory *pollerHistory
-		currentPolls  atomic.Int64
-		taskValidator taskValidator
+		pollerHistory              *pollerHistory
+		currentPolls               atomic.Int64
+		taskValidator              taskValidator
+		tasksAddedInIntervals      *taskTracker
+		tasksDispatchedInIntervals *taskTracker
 	}
 )
 
+// a circular array of a fixed size for tracking tasks
+type circularTaskBuffer struct {
+	buffer     []int
+	currentPos int
+}
+
+func newCircularTaskBuffer(size int) circularTaskBuffer {
+	return circularTaskBuffer{
+		buffer: make([]int, size),
+	}
+}
+
+func (cb *circularTaskBuffer) incrementTaskCount() {
+	cb.buffer[cb.currentPos]++
+}
+
+func (cb *circularTaskBuffer) advance() {
+	cb.currentPos = (cb.currentPos + 1) % len(cb.buffer)
+	cb.buffer[cb.currentPos] = 0 // Reset the task count for the new interval
+}
+
+// returns the total number of tasks in the buffer
```

Comment:
also not necessary, but optionally to be very Go-rule-following:

```suggestion
// totalTasks returns the total number of tasks in the buffer
```

I think GoLand only enforces function comment style for exported functions, so this is fine as is